lmd/datarow.go

```diff
@@ -746,3 +746,184 @@ func (d *DataRow) WriteJSON(json *jsoniter.Stream, columns *[]*Column) {
 	}
 	json.WriteRaw("]\n")
 }
+
+func (d *DataRow) isAuthorizedFor(authUser string, host string, service string) (canView bool) {
+	canView = false
+
+	var contacts []string
+	p := d.DataStore.Peer
+
+	// get contacts for host, if we are checking a host or
+	// if this is a service and ServiceAuthorization is loose
+	if (service != "" && p.LocalConfig.ServiceAuthorization == loose) ||
+		service == "" {
+		hostObj, ok := p.Tables["hosts"].Index[host]
+		contactsColumn := p.Tables["hosts"].GetColumn("contacts")
+		// Make sure the host we found is actually valid
+		if ok {
+			contacts = append(contacts, *hostObj.GetStringList(contactsColumn)...)
+		}
+
+	}
+
+	// get contacts on services
+	if len(service) > 0 {
+		var serviceID strings.Builder
+		serviceID.WriteString(host)
+		serviceID.WriteString(";")
+		serviceID.WriteString(service)
+		serviceObj, ok := p.Tables["services"].Index[serviceID.String()]
+		contactsColumn := p.Tables["services"].GetColumn("contacts")
+		if ok {
+			contacts = append(contacts, *serviceObj.GetStringList(contactsColumn)...)
+		}
+	}
+
+	for _, contact := range contacts {
```

Comment:
instead of creating a list just to iterate it once, wouldn't it be better to simply iterate over the contacts column and compare the authuser directly. This would also allow us to return early on a match.